src/apps/stable/features/playback/utils/mediaSegmentManager.ts

```diff
@@ -30,13 +30,28 @@ class MediaSegmentManager extends PlaybackSubscriber {
 
         try {
             const { data: mediaSegments } = await mediaSegmentsApi.getItemSegments({ itemId, includeSegmentTypes });
-            this.mediaSegments = mediaSegments.Items || [];
+            this.mediaSegments = this.adjustMediaSegments(mediaSegments.Items || []);
         } catch (err) {
             console.error('[MediaSegmentManager] failed to fetch segments', err);
             this.mediaSegments = [];
         }
     }
 
+    private adjustMediaSegments(segments: MediaSegmentDto[]): MediaSegmentDto[] {
```

Comment:
In order to decrease confusion, I wonder if this method should be renamed to `applySegmentSkipPreferences()` (or something along those lines, so that it's clear that you aren't adjusting the segment itself).